classes.py

```diff
@@ -0,0 +1,264 @@
+
+            del x[col]
+
+        return x
+
+    def get_feature_names(self):
+
+        self.feature_names
+
+
+
+class VarSelector(BaseEstimator,TransformerMixin):
+    
+    def __init__(self,feature_names):
+        
+        
+        self.feature_names=feature_names
+        
+    
+    def fit(self,x,y=None):
+
+        return self 
+    
+    def transform(self,x) :
+        
+        return x[self.feature_names]
+
+    def get_feature_names(self) :
+                
+        return self.feature_names
+
+
+class DataFrameImputer(BaseEstimator,TransformerMixin):
+    
+    def __init__(self):
+        
+        
+        self.feature_names=[]
+        self.impute_dict={}
+        
+    
+    def fit(self,x,y=None):
+        
+        self.feature_names=x.columns
+        
+        for col in x.columns:
+            
+            if x[col].dtype=='O':
+                self.impute_dict[col]='missing'
+            else :
+                self.impute_dict[col]=x[col].median()
+                
+        return self
+    
+    def transform(self,x) :
+        
+        return x.fillna(self.impute_dict)
+        
+    def get_feature_names(self) :
+                
+        return self.feature_names
+
+    
+class string_clean(BaseEstimator,TransformerMixin):
+    
+    def __init__(self,replace_it,replace_with):
+        
+        
+        self.feature_names=[]
+        self.replace_it=replace_it
+        self.replace_with=replace_with        
+    
+    def fit(self,x,y=None):
+
+        self.feature_names=x.columns
+
+        return self 
+    
+    def transform(self,x) :
+        
+        for col in x.columns:
+
+            x[col]=x[col].str.replace(self.replace_it,self.replace_with)
+        
+        return x
+
+    def get_feature_names(self) :
+                
+        return self.feature_names
+    
+    
+
+class count_vectorizer(BaseEstimator,TransformerMixin):
+
+    def __init__(self):
+        
+        
+        self.feature_names=[]
+        
+    
+    def fit(self,x,y=None):
+
+        self.feature_names=x.columns
+        
+        return self
+    
+    def transform(self,x) :
+        
+        for col in x.columns:
+            x[col]=x[col].TfidfVectorizer(analyzer='word',stop_words='english',max_df=0.8,min_df=0.01,max_features=200)
+            x[col]=x[col].toarray()
+            
+        return  x
+    
+    def get_feature_names(self) :
+                
+        return self.feature_names
+    
+    
+
+class standardise_num(BaseEstimator,TransformerMixin):
+    
+    
+    def __init__(self,feature_names):
+        
+        
+        self.feature_names=[]
+        
+    
+    def fit(self,x,y=None):
+        
+        self.feature_names=x.columns
+        
+        scaler = StandardScaler()
+
+        return self 
+    
+    def transform(self,x) :
+                      
+        scaled = scaler.fit_transform(x.columns)
+        
```

Comment:
dont use fit_transform . do fit under function fit and transform here . add `scaler` as an attribute in __init__ itself